R/zchunk_L2072.ag_water_irr_mgmt.R

```diff
@@ -1,67 +1,322 @@
+      # Calculate profit as commodity price minus water costs and non-land variable costs
+      mutate(Profit = calPrice - WaterCost - nonLandVariableCost) ->
+      L2072.AgCoef_IrrWaterWdraw_ag_mgmt
+
+    # Assume an exogenous floor on profit rates to prevent negative, zero, and very low profit rates
+    # For the model, low profit rates are fine as long as it's not the dominant crop in a nest where bioenergy is allowed
+    # Define minimum profit
+    minProfit <- min(L2072.AgCoef_IrrWaterWdraw_ag_mgmt["calPrice"] - L2072.AgCoef_IrrWaterWdraw_ag_mgmt["nonLandVariableCost"]) / 2
+    L2072.AgCoef_IrrWaterWdraw_ag_mgmt %>%
+      # Adjust profit floor
+      mutate(coefficient = round(pmin(coefficient, (calPrice - nonLandVariableCost - minProfit) / WaterPrice), aglu.DIGITS_CALOUTPUT)) %>%
+      # Remove the unnecessary columns
+      select(one_of(LEVEL2_DATA_NAMES[["AgCoef"]])) ->
+      L2072.AgCoef_IrrWaterWdraw_ag_mgmt
 
     # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
-      add_legacy_name("object") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
-      object
-
-    return_data(object)
+    L2072.AgCoef_IrrWaterCons_ag_mgmt %>%
+      add_title("Irrigation water consumption IO coefficients by region / irrigated crop / year / GLU / management level") %>%
+      add_units("km3/Mt") %>%
+      add_comments("The same IO coefficients are assigned to both high and low management and keep constant for all model years") %>%
+      add_legacy_name("L2072.AgCoef_IrrWaterCons_ag_mgmt") %>%
+      add_precursors("common/GCAM_region_names",
+                     "water/basin_to_country_mapping",
+                     "L165.BlueIrr_m3kg_R_C_GLU",
+                     "temp-data-inject/L161.ag_irrProd_Mt_R_C_Y_GLU",
+                     "water/A03.sector") ->
+      L2072.AgCoef_IrrWaterCons_ag_mgmt
+
+    L2072.AgCoef_IrrWaterWdraw_ag_mgmt %>%
+      add_title("Irrigation water withdrawals IO coefficients by region / irrigated crop / year / GLU / management level") %>%
+      add_units("km3/Mt") %>%
+      add_comments("Withdrawals coefs are calculated as consumption coefs divided by irrigation efficiency") %>%
+      add_comments("The same IO coefficients are assigned to both high and low management and keep constant for all model years") %>%
```

Comment:
Maybe mention that we set a floor on profit and adjust the coefficient to ensure that it is met?